presentation/frontend/src/presentation-frontend/StreamedResponseGenerator.ts

```diff
@@ -0,0 +1,156 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { concat, concatAll, map, mergeMap, Observable, of, range, scan } from "rxjs";
+import { eachValueFrom } from "rxjs-for-await";
+import { PagedResponse, PageOptions } from "@itwin/presentation-common";
+import { SortedArray } from "@itwin/core-bentley";
+import { MultipleValuesRequestOptions } from "./PresentationManager";
+
+/**
+ * Properties for streaming the results.
+ * @internal
+ */
+export type StreamedResponseGeneratorProps<TItem> = MultipleValuesRequestOptions & {
+  getBatch(page: Required<PageOptions>, requestIdx: number): Promise<PagedResponse<TItem>>;
+};
+
+/**
+ * This class allows loading values in multiple parallel batches and return them either as an array or an async iterator.
+ * Pages are prefetched in advanced according to the `parallelism` argument.
+ * @internal
+ */
+export class StreamedResponseGenerator<TPagedResponseItem> {
+  constructor(private readonly _props: StreamedResponseGeneratorProps<TPagedResponseItem>) {}
+
+  /** Creates a response with the total item count and an array of items for the requested page. */
+  public async createItemsResponse(): Promise<{ total: number; items: TPagedResponseItem[] }> {
```

Comment:
Looks like this is now used only in tests. Maybe the function should be moved there?

Reply:
I figured, this can be useful. I didn't use this method in `PresentationManager` so that I wouldn't have to write repeating tests for functions that return iterators.

Reply:
Removed